kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/config/Configuration.java

```diff
@@ -5,24 +5,118 @@
  */
 package io.kroxylicious.proxy.config;
 
+import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.Objects;
 import java.util.Optional;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
 
 import io.kroxylicious.proxy.config.admin.AdminHttpConfiguration;
 
 import edu.umd.cs.findbugs.annotations.NonNull;
 import edu.umd.cs.findbugs.annotations.Nullable;
 
-public record Configuration(@Nullable AdminHttpConfiguration adminHttp,
+/**
+ * The root of the proxy configuration.
+ * @param adminHttp
+ * @param filterDefinitions A list of named filter definitions (names must be unique)
+ * @param defaultFilters The names of the {@link #filterDefinitions()} to be use when a {@link VirtualCluster} doesn't specify its own {@link VirtualCluster#filterRefs()}.
+ * @param virtualClusters The virtual clusters
+ * @param filters Deprecated. The filter definitions to be used for all virtual clusters. Can only be specified if {@link #filterDefinitions()} is null.
+ * @param micrometer The micrometer config
+ * @param useIoUring true to use iouring
+ * @param development Development options
+ */
+@JsonPropertyOrder({ "adminHttp", "filterDefinitions", "defaultFilters", "virtualClusters", "filters", "micrometer", "useIoUring", "development" })
+public record Configuration(
+                            @Nullable AdminHttpConfiguration adminHttp,
+                            @Nullable List<NamedFilterDefinition> filterDefinitions,
+                            @Nullable List<String> defaultFilters,
                             Map<String, VirtualCluster> virtualClusters,
-                            List<FilterDefinition> filters,
+                            @Nullable List<FilterDefinition> filters,
```

Comment:
We should `@Deprecated` this field. Are we also able to deprecate the current canonical constructor?